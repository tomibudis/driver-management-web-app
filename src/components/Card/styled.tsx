import styled from "styled-components";
import { COLORS } from "~constants/index";

export const StyledCard = styled.div`
  background: white;
  width: 250px;
  border-radius: 4px;

  > div:nth-child(1) {
    display: flex;
    justify-content: space-between;
    > div > span {
      color: ${COLORS.RED};
    }
  }

  > hr {
    height: 2px;
    opacity: 0.1;
    padding: 0px;
    margin: 0px;
  }
`;
