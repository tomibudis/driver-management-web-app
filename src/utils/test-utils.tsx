import React from "react";
import { configureStore, EnhancedStore } from "@reduxjs/toolkit"; // for redux-toolkit
import {
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import driversReducer from "../redux/drivers/reducer";

import { RootState } from "../redux/store";
import mockDriversJson from "../views/driver-management/__test__/mock-drivers.json";

type ReduxRenderOptions = {
  preloadedState?: RootState;
  store?: EnhancedStore;
  renderOptions?: Omit<RenderOptions, "wrapper">;
};

function render(
  ui: ReactElement,
  {
    preloadedState = {
      drivers: {
        results: mockDriversJson.results,
        data: mockDriversJson?.results,
      },
    },
    store = configureStore({
      reducer: { drivers: driversReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ReduxRenderOptions = {}
): RenderResult {
  function Wrapper({ children }: { children?: ReactNode }): ReactElement {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
