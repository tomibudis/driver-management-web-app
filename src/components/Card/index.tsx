import React from "react";
import { DEFAULT_EMPTY } from "~constants/index";
import useResponsive from "~src/hooks/useResponsive";

import { StyledCard } from "./styled";
interface CardProps {
  driverId: string;
}
const Card: React.FC<CardProps> = ({ children, driverId }) => {
  const { isMobile } = useResponsive();
  return (
    <StyledCard isMobile={isMobile}>
      <div className="p-3">
        <div>
          Driver ID: <span>{driverId || DEFAULT_EMPTY}</span>
        </div>
        <i className="bi bi-three-dots" />
      </div>

      <hr />
      <div className="p-3">{children}</div>
    </StyledCard>
  );
};

export default Card;
