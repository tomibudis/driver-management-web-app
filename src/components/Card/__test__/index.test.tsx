import React from "react";
import { render, screen } from "~utils/test-utils";

import Card from "../index";

describe("<Card />", () => {
  it("render card component", () => {
    render(<Card driverId="1234">Testing</Card>);

    const children = screen.getByText("Testing");
    expect(children).toBeInTheDocument();
  });
  it("render driver id correctly", () => {
    render(<Card driverId="1234">Testing</Card>);
    const driverId = screen.getByText("1234");
    expect(driverId).toBeInTheDocument();
  });
  it("matched with a snapshot", () => {
    const { container } = render(<Card driverId="1234">Testing</Card>);
    expect(container).toMatchSnapshot();
  });
});
