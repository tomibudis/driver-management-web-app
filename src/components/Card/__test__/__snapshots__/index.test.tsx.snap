// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matched with a snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ gksoUO"
  >
    <div
      class="p-3"
    >
      <div>
        Driver ID: 
        <span>
          1234
        </span>
      </div>
      <i
        class="bi bi-three-dots"
      />
    </div>
    <hr />
    <div
      class="p-3"
    >
      Testing
    </div>
  </div>
</div>
`;
